### ISSUE TITLE
Pagination may not work correctly for archived workflows

### ISSUE THREAD

**Issue by terrytangyuan** [argoproj/argo-workflows#11715]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

When we perform the merge of live workflows and archived workflows (in order to only show one of them in the UI), the `listMeta`  in `WorkflowList` is obtained from live workflows. This may cause issues for pagination for archived workflows.

```go
mergedWfsList := v1alpha1.WorkflowList{Items: mergedWfs, ListMeta: liveWfs.ListMeta}
```

We need to test the pagination more thoroughly and revisit this part of the code. https://github.com/argoproj/argo-workflows/blob/master/server/workflow/workflow_server.go#L155C2-L155C86

### Version

latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
TBA
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

**Comment by terrytangyuan** [comment]
@sunyeongchoi  Would you like to help take a look at this? Your recent PRs on pagination (https://github.com/argoproj/argo-workflows/pull/11703, https://github.com/argoproj/argo-workflows/pull/11684) brought my attention to this part of the code.

**Comment by sunyeongchoi** [comment]
@terrytangyuan 
Of course!! I will analyze the issue at hand :)

**Comment by sunyeongchoi** [comment]
@terrytangyuan 

Hello. I would like to share the results of my analysis on the issue so far.

```go
func (s *workflowServer) ListWorkflows(ctx context.Context, req *workflowpkg.WorkflowListRequest) (*wfv1.WorkflowList, error) {
	wfClient := auth.GetWfClient(ctx)

	listOption := &metav1.ListOptions{}
	if req.ListOptions != nil {
		listOption = req.ListOptions
	}
	s.instanceIDService.With(listOption)
	wfList, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).List(ctx, *listOption)
	if err != nil {
		return nil, sutils.ToStatusError(err, codes.Internal)
	}

// In this part, always return err. Because archived workflows use int type continue not string type(workflows use).
	archivedWfList, err := s.wfArchiveServer.ListArchivedWorkflows(ctx, &workflowarchivepkg.ListArchivedWorkflowsRequest{
		ListOptions: listOption,
		NamePrefix:  "",
		Namespace:   req.Namespace,
	})

	if err != nil {
		log.Warnf("unable to list archived workflows:%v", err)
	} else {
		if archivedWfList != nil {
			wfList = mergeWithArchivedWorkflows(*wfList, *archivedWfList, int(listOption.Limit))
		}
	}
...
```
In the given code, we cannot fetch data unconditionally in ListArchivedWorkflows. This is because the continue value used in ListArchivedWorkflows is of type int, whereas in Workflows, it's a string. As a result, it always leads to an error.

This error is being logged as follows:
`server: time="2023-09-03T01:02:29.487Z" level=warning msg="unable to list archived workflows:rpc error: code = InvalidArgument desc = listOptions.continue must be int"`

In essence, this problem arises because, as mentioned earlier, Workflows and ArchivedWorkflows use different pagination methods, and the issue occurs when we try to use the list options from Workflows in ArchivedWorkflows.

It would be a good idea for us to discuss how to address this issue together.
After considering various solutions, I will share them with you shortly. :)

**Comment by sunyeongchoi** [comment]
@terrytangyuan 

I have one question.

After version `v3.4.10`, there used to be a separate "archived workflows" page, but I'm curious why the current `master` branch does not have the "archived workflows" page.

**Comment by terrytangyuan** [comment]
> Workflows and ArchivedWorkflows use different pagination methods, and the issue occurs when we try to use the list options from Workflows in ArchivedWorkflows.

Yes that's exactly what the issue is! Let me know if you can come up with solution to merge the two lists. 

> After version v3.4.10, there used to be a separate "archived workflows" page, but I'm curious why the current master branch does not have the "archived workflows" page.

Yes, we removed the archived workflows page and both types of workflows are now displayed in one list view.

**Comment by agilgur5** [comment]
Re-opening this since #12068 reverted #11761

**Comment by agilgur5** [comment]
This should be re-resolved by #12736 now
